pkg/controller/dns/dns_policy.go

```diff
@@ -0,0 +1,204 @@
+		managed.WithPollInterval(o.PollInterval),
+		managed.WithLogger(o.Logger.WithValues("controller", name)),
+		managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name))),
+		managed.WithConnectionPublishers(cps...))
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(o.ForControllerRuntime()).
+		For(&v1alpha1.Policy{}).
+		Complete(ratelimiter.NewReconciler(name, r, o.GlobalRateLimiter))
+}
+
+// Connector is ..
+type Connector struct {
+	kube client.Client
+}
+
+// Connect is
+func (c *Connector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	projectID, opts, err := gcp.GetConnectionInfo(ctx, c.kube, mg)
+	if err != nil {
+		return nil, err
+	}
+
+	d, err := dns.NewService(ctx, opts...)
+	if err != nil {
+		return nil, errors.Wrap(err, errNewClient)
+	}
+	return &policyExternal{
+		kube:      c.kube,
+		dns:       d.Policies,
+		projectID: projectID,
+	}, nil
+}
+
+type policyExternal struct {
+	kube      client.Client
+	dns       *dns.PoliciesService
+	projectID string
+}
+
+func (e *policyExternal) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mg.(*v1alpha1.Policy)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errorNotDNSPolicy)
+	}
+
+	policy, err := e.dns.Get(
+		e.projectID,
+		meta.GetExternalName(cr),
+	).Context(ctx).Do()
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(
+			resource.Ignore(gcp.IsErrorNotFound, err),
+			errorGetFailed,
+		)
+	}
+
+	cr.SetConditions(xpv1.Available())
+
+	UpToDate, err := dnsclient.IssUpToDate(
+		meta.GetExternalName(cr),
+		&cr.Spec.ForProvider,
+		policy)
+
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errCheckUpToDate)
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists:   true,
+		ResourceUpToDate: UpToDate,
+	}, nil
+
+}
+
+func (e *policyExternal) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mg.(*v1alpha1.Policy)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errorNotDNSPolicy)
+	}
+
+	args := &dns.Policy{}
+
+	dnsclient.GenerateDNSPolicy(
+		meta.GetExternalName(cr),
+		cr.Spec.ForProvider,
+		args,
+	)
+
+	_, err := e.dns.Create(
+		e.projectID,
+		args,
+	).Context(ctx).Do()
+	return managed.ExternalCreation{}, errors.Wrap(err, errorCreatePolicy)
+}
+
+func (e *policyExternal) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mg.(*v1alpha1.Policy)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errorNotDNSPolicy)
+	}
+
+	args := &dns.Policy{}
+	dnsclient.GenerateDNSPolicy(
+		meta.GetExternalName(cr),
+		cr.Spec.ForProvider,
+		args,
+	)
+	_, err := e.dns.Patch(
+		e.projectID,
+		meta.GetExternalName(cr),
+		args,
+	).Context(ctx).Do()
+	if err != nil {
+		return managed.ExternalUpdate{}, err
+	}
+	return managed.ExternalUpdate{}, nil
```

Comment:
```suggestion
	return managed.ExternalUpdate{}, errors.Wrap(err, errCannotUpdate)
```

This could be easier to follow.